src/Agent.Worker/Build/BuildJobExtension.cs

```diff
@@ -212,7 +215,10 @@ private void UpdateCheckoutTasksAndVariables(IExecutionContext executionContext,
                 if (hasOnlyOneCheckoutTask || RepositoryUtil.IsPrimaryRepositoryName(repositoryAlias))
                 {
                     submoduleCheckout = checkoutTask.Inputs.ContainsKey(PipelineConstants.CheckoutTaskInputs.Submodules);
-                    repoClean = repoClean ?? checkoutTask.Inputs.ContainsKey(PipelineConstants.CheckoutTaskInputs.Clean);
+                    if (checkoutTask.Inputs.TryGetValue(PipelineConstants.CheckoutTaskInputs.Clean, out string cleanInputValue))
+                    {
+                        repoClean = Boolean.TryParse(cleanInputValue, out bool cleanValue) ? cleanValue : true;
```

Comment:
Previously repoClean was assigned here only if RepoClean variable has not been specified, but currently it could be assigned by clean value for primary repo, and used further.
So for example for next tasks:
- checkout: self
   clean: true
- checkout : other-repo

as I see it will use clean parameter (true) for second checkout also - is this expected?

Reply:
Regarding the first part of your comment, I agree, I've missed this condition and going to add it.
But regarding your example it is an expected behavior since default value of clean parameter is true, you can check it in task [definition](https://mseng.visualstudio.com/AzureDevOps/_git/AzureDevOps?path=%2FDistributedTask%2FClient%2FWebApi%2FPipelines%2FPipelineConstants.cs&version=GBmaster&line=150&lineEnd=151&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents)

In fact this missed condition doesn't affect task behavior, because if the Clean value weren't set from the server Task's clean option won't be overwritten because of newly added flag [here](https://github.com/microsoft/azure-pipelines-agent/pull/3274/files#diff-9621cecbf92d9b77b3c794786578b90c290e60087aa323ea8b01daeb9cc1a97cR205), if the clean option was set on the server then Clean input of Self checkout task will be overwritten by [this function](https://github.com/microsoft/azure-pipelines-agent/pull/3274/files#diff-9621cecbf92d9b77b3c794786578b90c290e60087aa323ea8b01daeb9cc1a97cR205), if the clean option was set on the server then Clean input of Self checkout task will be overwritten by this function and in the block that you mentioned we will just reassign the same value to the repoClean.

Reply:
Added missed condition. Thank you!